io/disk/bonnie.py

```diff
@@ -89,20 +99,62 @@ def setUp(self):
         self.data_size = self.params.get('data_size_to_pass', default=0)
 
         if self.disk is not None:
-            self.part_obj = Partition(self.disk, mountpoint=self.scratch_dir)
-            self.log.info("Test will run on %s", self.scratch_dir)
-            self.log.info("Unmounting disk/dir before creating file system")
-            self.part_obj.unmount()
-            self.log.info("creating %s file system on %s disk",
-                          fstype, self.disk)
-            self.part_obj.mkfs(fstype)
-            self.log.info("Mounting disk %s on directory %s",
-                          self.disk, self.scratch_dir)
-            try:
-                self.part_obj.mount()
-            except PartitionError:
-                self.fail("Mounting disk %s on directory %s failed"
-                          % (self.disk, self.scratch_dir))
+            if self.disk in disk.get_disks():
+                if raid_needed:
+                    raid_name = '/dev/md/mdsraid'
+                    self.create_raid(self.disk, raid_name)
+                    self.raid_create = True
+                    self.disk = raid_name
```

Comment:
@smruti77 We need to add `self.scratch_dir = self.disk` here as well

Reply:
ok